Assets/Scripts/BaseTurret.cs

```diff
@@ -19,13 +19,23 @@ public abstract class BaseTurret : MonoBehaviour
     [SerializeField] protected SpriteRenderer spriteRenderer;
     [SerializeField] protected TurretUpgrades upgrades;
     [SerializeField] protected GameObject upgradesMenu;
-
+    private RectTransform upgradesMenuRectTransform;
```

Comment:
```suggestion
    [SerializeField] private RectTransform upgradesMenuRectTransform;
```
Można tak dodać też do pozostałych pół, żeby się łatwiej debugowalo, np. referencje pod headerem "References"